pkg/domain/domain.go

```diff
@@ -1808,6 +1810,25 @@ func (do *Domain) GetPDHTTPClient() pdhttp.Client {
 	return nil
 }
 
+func decodePrivilegeEvent(resp clientv3.WatchResponse) PrivilegeEvent {
+	var msg PrivilegeEvent
+	for _, event := range resp.Events {
+		if event.Kv != nil {
+			val := event.Kv.Value
+			if len(val) > 0 {
+				err := json.Unmarshal(val, &msg)
+				if err == nil {
+					break
+				}
+			}
+		}
+	}
+	if len(msg.UserList) == 0 {
```

Comment:
Any case that `len(UserList) == 0 and All == false` and we should still update all privileges? If not, I think it's better to remove this logic.

Reply:
When something is wrong, `len(msg.UserList) == 0`
Then we fall back to the old reload all way.